packages/fxa-auth-server/lib/payments/processing-tasks-shared.ts

```diff
@@ -0,0 +1,87 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import { setupAuthDatabase } from 'fxa-shared/db';
+import { StatsD } from 'hot-shots';
+import Container from 'typedi';
+
+import error from '../error';
+import { CurrencyHelper } from '../payments/currencies';
+import { PayPalHelper } from './paypal';
+import { PayPalClient } from './paypal-client';
+import { StripeHelper } from './stripe';
+import { configureSentry } from '../sentry';
+
+const config = require('../../config').getProperties();
+
+export default async function initShared(isPaypalProcessor = true) {
```

Comment:
I'm not a fan of this type of toggle params.  Since it's a boolean, when we have another script that needs additional setup, do we 1) add another boolean parameter, 2) change it to a sum type, or 3) change it to an enum (or something else)?

I understand the need for it though.  What do you think of accepting a function that allows further setup needed by the calling script?  That way this shared lib need not handle whatever's wanted by the calling scripts.

Reply:
We should try and avoid default exports unless we need to interact with legacy code that assumes every module exports a function.

Reply:
> What do you think of accepting a function that allows further setup needed by the calling script?

Sure that sounds good. I modified this with the diff below:
```diff
diff --git a/packages/fxa-auth-server/lib/payments/processing-tasks-shared.ts b/packages/fxa-auth-server/lib/payments/processing-tasks-shared.ts
index 3fdd02624..e0935619f 100644
--- a/packages/fxa-auth-server/lib/payments/processing-tasks-shared.ts
+++ b/packages/fxa-auth-server/lib/payments/processing-tasks-shared.ts
@@ -7,14 +7,12 @@ import Container from 'typedi';
 
 import error from '../error';
 import { CurrencyHelper } from '../payments/currencies';
-import { PayPalHelper } from './paypal';
-import { PayPalClient } from './paypal-client';
 import { StripeHelper } from './stripe';
 import { configureSentry } from '../sentry';
 
 const config = require('../../config').getProperties();
 
-export default async function initShared(isPaypalProcessor = true) {
+export async function initShared(additionalSetupFn?: Function) {
   configureSentry(undefined, config);
   // Establish database connection and bind instance to Model using Knex
   setupAuthDatabase(config.database.mysql.auth);
@@ -70,13 +68,8 @@ export default async function initShared(isPaypalProcessor = true) {
   const stripeHelper = new StripeHelper(log, config, statsd);
   Container.set(StripeHelper, stripeHelper);
 
-  if (isPaypalProcessor) {
-    const paypalClient = new PayPalClient(
-      config.subscriptions.paypalNvpSigCredentials
-    );
-    Container.set(PayPalClient, paypalClient);
-    const paypalHelper = new PayPalHelper({ log });
-    Container.set(PayPalHelper, paypalHelper);
+  if (additionalSetupFn) {
+    additionalSetupFn(log);
   }
 
   return {
diff --git a/packages/fxa-auth-server/scripts/paypal-processor.ts b/packages/fxa-auth-server/scripts/paypal-processor.ts
index 4b4ece454..d3ec7d111 100644
--- a/packages/fxa-auth-server/scripts/paypal-processor.ts
+++ b/packages/fxa-auth-server/scripts/paypal-processor.ts
@@ -3,12 +3,26 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 import program from 'commander';
 
-import initShared from '../lib/payments/processing-tasks-shared';
+import Container from 'typedi';
+import { Logger } from 'mozlog';
+import { PayPalHelper } from '../lib/payments/paypal';
+import { PayPalClient } from '../lib/payments/paypal-client';
+
+import { initShared } from '../lib/payments/processing-tasks-shared';
 import { PaypalProcessor } from '../lib/payments/paypal-processor';
 
 const pckg = require('../package.json');
 const config = require('../config').getProperties();
 
+function additionalSetup(log: Logger) {
+  const paypalClient = new PayPalClient(
+    config.subscriptions.paypalNvpSigCredentials
+  );
+  Container.set(PayPalClient, paypalClient);
+  const paypalHelper = new PayPalHelper({ log });
+  Container.set(PayPalHelper, paypalHelper);
+}
+
 export async function init() {
   // Load program options
   program
@@ -26,7 +40,7 @@ export async function init() {
     )
     .parse(process.argv);
 
-  const { log, database, senders } = await initShared();
+  const { log, database, senders } = await initShared(additionalSetup);
 
   const processor = new PaypalProcessor(
     log,
diff --git a/packages/fxa-auth-server/scripts/subscription-reminders.ts b/packages/fxa-auth-server/scripts/subscription-reminders.ts
index df042edeb..7156d8b2e 100644
--- a/packages/fxa-auth-server/scripts/subscription-reminders.ts
+++ b/packages/fxa-auth-server/scripts/subscription-reminders.ts
@@ -4,7 +4,7 @@
 
 import program from 'commander';
 
-import initShared from '../lib/payments/processing-tasks-shared';
+import { initShared } from '../lib/payments/processing-tasks-shared';
 import { SubscriptionReminders } from '../lib/payments/subscription-reminders';
 
 const pckg = require('../package.json');
@@ -29,7 +29,7 @@ async function init() {
     .parse(process.argv);
 
   const isPaypalProcessor = false;
-  const { log, database, senders } = await initShared(isPaypalProcessor);
+  const { log, database, senders } = await initShared();
 
   const subscriptionReminders = new SubscriptionReminders(
     log,

```

Reply:
Barry and I discussed this sync and resolved it.